libraries/aws/common/src/aws_iot_doc_parser.c

```diff
@@ -32,77 +32,78 @@
 #include <string.h>
 
 /* JSON utilities include. */
-#include "iot_json_utils.h"
+#include "aws_iot_doc_parser.h"
 
 #define IS_QUOTE( str, idx ) \
-    ( str[ idx ] == '"' && ( idx == 0 || str[ idx - 1 ] != '\\' ) )
+    ( ( str )[ ( idx ) ] == '"' && ( ( idx ) == 0 || ( str )[ ( idx ) - 1 ] != '\\' ) )
 #define IS_WHITESPACE( str, idx ) \
-    ( str[ idx ] == ' ' || str[ idx ] == '\n' || str[ idx ] == '\r' || str[ idx ] == '\t' )
+    ( ( str )[ ( idx ) ] == ' ' || ( str )[ ( idx ) ] == '\n' || ( str )[ ( idx ) ] == '\r' || ( str )[ ( idx ) ] == '\t' )
 
 /*-----------------------------------------------------------*/
 
-bool IotJsonUtils_FindJsonValue( const char * pJsonDocument,
-                                 size_t jsonDocumentLength,
-                                 const char * pJsonKey,
-                                 size_t jsonKeyLength,
-                                 const char ** pJsonValue,
-                                 size_t * pJsonValueLength )
+bool AwsIotDocParser_FindValue( const char * pAwsIotJsonDocument,
```

Comment:
This function has a high [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and would be a good candidate for refactoring.  As written, this is very difficult to review.

```sh
$ pmccabe -v libraries/standard/serializer/src/iot_json_utils.c 
Modified McCabe Cyclomatic Complexity
|   Traditional McCabe Cyclomatic Complexity
|       |    # Statements in function
|       |        |   First line of function
|       |        |       |   # lines in function
|       |        |       |       |  filename(definition line number):function
|       |        |       |       |           |
39	41	72	44	234	libraries/standard/serializer/src/iot_json_utils.c(44): IotJsonUtils_FindJsonValue
```